src/zip_clones.rs

```diff
@@ -0,0 +1,77 @@
+/// An iterator which iterate over an iterator and clones of a value.
+///
+///
+/// See [`.zip_clones()`](crate::Itertools::zip_clones) for more information.
+pub struct ZipClones<I, T>
+where
+    I: Iterator,
+{
+    iter: I,
+    next: Option<I::Item>,
+    zipped: Option<T>,
+}
+
+/// Zips an iterator with clones of a value.
+///
+/// [`IntoIterator`] enabled version of [`Itertools::zip_clones`](crate::Itertools::zip_clones).
+///
+/// ```
+/// use itertools::zip_clones;
+///
+/// let data = [1, 2, 3, 4, 5];
+/// let zipped = "expensive-to-clone".to_string();
+/// for (a, b) in zip_clones(&data, zipped) {
+///     // do something that consumes the expensive zipped value
+/// }
+/// ```
+pub fn zip_clones<I, T>(i: I, zipped: T) -> ZipClones<I::IntoIter, T>
+where
+    I: IntoIterator,
+    T: Clone,
+{
+    let mut iter = i.into_iter();
+    let next = iter.next();
```

Comment:
In general, I dont think so, because you need to know if the current call to next should clone the zipped value or consume it, so you must peek.
We can maybe avoid the peek during the creation of the iterator, but not the peeks during `next()` calls. tell me if you think it has any value.